iothub/device/src/IotHubClientOptions.cs

```diff
@@ -112,6 +112,7 @@ internal IotHubClientOptions Clone()
                 ModelId = ModelId,
                 SdkAssignsMessageId = SdkAssignsMessageId,
                 AdditionalUserAgentInfo = AdditionalUserAgentInfo,
+                RetryPolicy = RetryPolicy,
```

Comment:
Previously we didn't pass in the user-supplied `RetryPolicy` here, which caused the hub device/module client can only use the default retry policy rather than setting with their own choices.

Reply:
Our test for this may need to use recursion to find properties that weren't cloned, otherwise we rely only on PRs to validate.